### PR TITLE
Fix always expanded "More" dropdown menu

### LINKED ISSUE #2357: Bug: "More" dropdown always expanded
Refined GitHub enabled on Chrome:

image

Disabled:

image

It just started today, and I've only noticed it does this when viewing a Repo (not other views afaik).

Clicking the dropdown, or anything, doesn't hide the dropdown. Seems like it's just stuck in this state.

### PR DESCRIPTION
**Pull request by kidonng**
Closes #2357
Closes #2358

For some reason using `<button>` won't open the dropdown, I have read GitHub's code for some time but can't figure out why, so changed it to `<span>` and everything looks fine.

# Test

This repo

### CODE REVIEW

**Comment by yakov116** [comment]
@sindresorhus 	@fregante 	please merge...

**Comment by casz** [comment]
@kidonng you could add closes #2358 😆

**Comment by kidonng** [comment]
@casz Oh I guess I can only close it when creating the PR but never mind

**Comment by casz** [comment]
GitHub is finicky you have to preface it with closes https://help.github.com/en/articles/closing-issues-using-keywords#closing-multiple-issues
```
Closes #2357 closes #2358
```

**Comment by kidonng** [comment]
@casz Didn't know that, thanks for the information!

**Comment by truongnmt** [comment]
Why we have to change from `button` to `summary`? 🤔

**Comment by matschaffer** [comment]
Did an unpacked build (per https://github.com/sindresorhus/refined-github/blob/master/contributing.md) from this PR and seems to be working great so far

**Comment by yolopunk** [comment]
> Why we have to change from `button` to `summary`? 🤔

Absolutely, this is over design.

**Comment by matschaffer** [comment]
@yolopunk 	@truongnmt 	

> For some reason using <button> won't open the dropdown, I have read GitHub's code for some time but can't figure out why, so changed it to <span> and everything looks fine.

Pretty sure @kidonng 	meant `<summary>` there, not `<span>`

File: source/features/more-dropdown.tsx [hunk 1]
```diff
@@ -13,15 +13,15 @@ const repoUrl = getRepoURL();
...
-			<button type="button" className="btn-link reponav-item js-menu-target" aria-expanded="false" aria-haspopup="true">
-				{'More '}
-				<span className="dropdown-caret"/>
-			</button>
-			<div className="dropdown-menu-content js-menu-content">
-				<div className="dropdown-menu dropdown-menu-se"/>
-			</div>
-		</div>
+		<details className="reponav-dropdown details-overlay details-reset">
+			<summary>
```
Comment (mobilutz) [inline source/features/more-dropdown.tsx]: I think the `btn btn-link` classes can also go onto the `<summary>`. At least this is how i.e. the `Review changes` looks like.

File: source/features/more-dropdown.tsx [hunk 2]
```diff
@@ -13,15 +13,15 @@ const repoUrl = getRepoURL();
...
+				<span className="btn-link reponav-item">
```
Comment (loilo) [inline source/features/more-dropdown.tsx]: I think a `role="button"` wouldn't hurt here — any a11y experts out there to correct me if I'm wrong on this?

**Comment by kidonng** [comment]
https://github.githubassets.com/assets/github-bootstrap-4b4ca38b.js

Formatted via Chrome, line 916-1031

```js
class DetailsMenuElement extends HTMLElement {
        constructor() {
          super()
        }

        get preload() {
          return this.hasAttribute("preload")
        }

        set preload(e) {
          e ? this.setAttribute("preload", "") : this.removeAttribute("preload")
        }

        get src() {
          return this.getAttribute("src") || ""
        }

        set src(e) {
          this.setAttribute("src", e)
        }

        connectedCallback() {
          this.setAttribute("role", "menu");
          const e = this.parentElement;
          if (!e)
            return;
          const t = e.querySelector("summary");
          t && t.setAttribute("aria-haspopup", "menu"),
          e.addEventListener("click", Qt),
          e.addEventListener("change", Qt),
          e.addEventListener("keydown", en),
          e.addEventListener("toggle", Vt, {
            once: !0
          }),
          e.addEventListener("toggle", Yt),
          this.preload && e.addEventListener("mouseover", Vt, {
            once: !0
          });
          const n = [Kt(e)];
          zt.set(this, {
            details: e,
            subscriptions: n,
            loaded: !1
          })
        }

        disconnectedCallback() {
          const e = zt.get(this);
          if (!e)
            return;
          zt.delete(this);
          const t = e.details
            , n = e.subscriptions;
          for (const s of n)
            s.unsubscribe();
          t.removeEventListener("click", Qt),
          t.removeEventListener("change", Qt),
          t.removeEventListener("keydown", en),
          t.removeEventListener("toggle", Vt, {
            once: !0
          }),
          t.removeEventListener("toggle", Yt),
          t.removeEventListener("mouseover", Vt, {
            once: !0
          })
        }
      }

      const zt = new WeakMap;

      function Vt(e) {
        const t = e.currentTarget;
        if (!(t instanceof Element))
          return;
        const n = t.querySelector("details-menu");
        if (!n)
          return;
        const s = n.getAttribute("src");
        if (!s)
          return;
        const o = zt.get(n);
        if (!o)
          return;
        if (o.loaded)
          return;
        o.loaded = !0;
        const i = n.querySelector("include-fragment");
        i && !i.hasAttribute("src") && (i.addEventListener("loadend", ()=>Xt(t)),
        i.setAttribute("src", s))
      }

      function Kt(e) {
        let t = !1;
        const n = ()=>t = !0
          , s = ()=>t = !1
          , o = ()=>{
          Xt(e),
          e.hasAttribute("open") && !t && function(e) {
            const t = document.activeElement;
            if (t && tn(t) && e.contains(t))
              return;
            const n = Jt(e, !0);
            n && n.focus()
          }(e)
        }
        ;
        return e.addEventListener("mousedown", n),
        e.addEventListener("keydown", s),
        e.addEventListener("toggle", o),
        {
          unsubscribe: ()=>{
            e.removeEventListener("mousedown", n),
            e.removeEventListener("keydown", s),
            e.removeEventListener("toggle", o)
          }
        }
      }

      function Yt(e) {
        const t = e.currentTarget;
        if (t instanceof Element && t.hasAttribute("open"))
          for (const n of document.querySelectorAll("details[open] > details-menu")) {
            const e = n.closest("details");
            e && e !== t && !e.contains(t) && e.removeAttribute("open")
          }
      }
```

**Comment by kidonng** [comment]
@truongnmt @yolopunk @matschaffer I didn't change `<button>` to `<summary>`, I wrapped the `<button>` with `<summary>` (which is needed for the dropdown to work, see the code above) but found that for some reason it didn't work, so I changed the `<button>` to `<span>` instead.

File: source/features/more-dropdown.tsx (hunk 1 above)
Comment (kidonng) [reply source/features/more-dropdown.tsx]: Absolutely!

File: source/features/more-dropdown.tsx (hunk 2 above)
Comment (kidonng) [reply source/features/more-dropdown.tsx]: I don't quite know about how to properly apply a11y here, but GitHub will add `aria-haspopup="menu"` to `<summary>` and other dropdowns don't have extra attributes so I guess it's fine.

**Comment by leggsimon** [comment]
@ChristianVermeulen comments like that aren’t really helpful. If it’s bothering you that much you can disable this feature by right clicking on the Refined Github extension go to Options, scroll down and uncheck the “more-dropdown” feature.

**Comment by yakov116** [comment]
@busches 	can you merge it?

**Comment by yakov116** [comment]
@busches 	can you run hotfix from travis?

**Comment by busches** [comment]
> @busches can you run hotfix from travis?

Not to my knowledge, need @fregante or @sindresorhus for that.

**Comment by notlmn** [comment]
Thanks for the fix @kidonng.

-----------------

> @busches can you run hotfix from travis?

Triggered a new build on Travis, AMO updated with `19.8.21` and the same should be reflected on Chrome Web Store soon.

**Comment by casz** [comment]
I was able to update the extension. Go here `chrome://extensions/?id=hlepfoohegkhhmjieoechaddaejaokhf` and click update.

**Comment by fregante** [comment]
Thanks all for the quick fix. Sindre and I are both AFK

**Comment by fregante** [comment]
@busches I think all collaborators can push a hotfix on Travis by triggering a manual build.

**Comment by busches** [comment]
@fregante, ah I see the option now, TIL.

**Pull request merged**